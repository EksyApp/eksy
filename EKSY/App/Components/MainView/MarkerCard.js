import React, {Component} from 'react';
import {
	View,
	StyleSheet,
	TouchableWithoutFeedback
} from 'react-native';
import {Card, CardSection, Label, TextArea} from '../Common/index'
import Image from 'react-native-fast-image'

export default class MarkerCard extends Component {

	handlePress() {
		this.props.setMarkerSelected(this.props.marker)
		this.props.setMarkerViewVisible()
		this.props.disableGestures(true)
	}

	renderWithImage() {
		return (
				<Card style={[styles.container, {width: this.props.width}]}>
					<CardSection style={[styles.imageContainer, {width: this.props.width}]}>
						<Image resizeMode={Image.resizeMode.cover} style={styles.image} source={{uri: this.props.marker.images[0].uri}}/>
					</CardSection>
					<View style={styles.infoContainer}>
						<Label style={styles.title}>{this.props.marker.title}</Label>
					</View>
				</Card>
		)

	}

	renderWithoutImage() {
		return (
				<Card style={[styles.container, {width: this.props.width}]}>
					<CardSection style={styles.infoContainer}>
						<Label style={styles.title}>{this.props.marker.title}</Label>
					</CardSection>

					<TextArea>{this.props.marker.text}</TextArea>

				</Card>
		)
	}

	renderCard() {
		if (this.props.marker.images && this.props.marker.images.length > 0) {
			return this.renderWithImage()
		} else {
			return this.renderWithoutImage()
		}
	}

	render() {
		return(
				<TouchableWithoutFeedback onPress={() => {this.handlePress()}}>
					{this.renderCard()}
				</TouchableWithoutFeedback>
		)
	}

}

const styles = StyleSheet.create({
	container: {
		height: 200,
		marginLeft: 0,
	},

	imageContainer: {
		width: "100%",
		height: "70%",
		padding: 0
	},

	image: {
		height: "100%",
		width: "100%"
	},

	infoContainer: {
		width: "100%",
		alignItems: "center",
		justifyContent: "center",
		overflow: "scroll"
	},

	title: {
		marginTop: 10
	}

});
