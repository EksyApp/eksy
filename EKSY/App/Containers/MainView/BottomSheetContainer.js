import React, {Component} from 'react'
import BottomSheetComponent from "../../Components/MainView/BottomSheetComponent";
import {connect} from "react-redux";
import * as ReduxActions from "../../Actions";

class BottomSheetContainer extends Component {
	
	render() {
		return (
			<BottomSheetComponent
				markerList={this.props.markerList}
			  setMarkerSelected={this.props.setMarkerSelected}
			  setMarkerViewVisible={this.props.setMarkerViewVisible}
			  disableGestures={this.props.disableGestures}
			/>
		)
	}
	
}

const mapStateToProps = (state) => {
	return {
		markerList: state.markers.markerList,
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		setMarkerSelected: (marker) => {
			dispatch(ReduxActions.setMarkerSelected(marker))
		},
		setMarkerViewVisible: () => {
			dispatch(ReduxActions.setMarkerViewVisible())
		},
		disableGestures: (value) => {
			dispatch(ReduxActions.disableGestures(value))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(BottomSheetContainer)