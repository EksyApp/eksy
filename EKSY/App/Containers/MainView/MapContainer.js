import React, {Component} from 'react'
import * as ReduxActions from "../../Actions/";
import MapComponent from "../../Components/MainView/MapComponent";
import {connect} from "react-redux";

export class MapContainer extends Component {
	
	render() {
		return (
				<MapComponent
						currentRegion={this.props.currentRegion}
						regionChange={this.props.regionChange}
						markerList={this.props.markerList}
						setMarkerSelected={this.props.setMarkerSelected}
						setMarkerViewVisible={this.props.setMarkerViewVisible}
						disableGestures={this.props.disableGestures}
						currentLocation={this.props.currentLocation}
				/>
		)
	}
	
}

const mapStateToProps = (state) => {
	return {
		currentRegion: state.map.currentRegion,
		currentLocation: state.map.location,
		markerList: state.markers.markerList,
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		regionChange: (region) => {
			dispatch(ReduxActions.updateRegion(region))
		},
		setMarkerSelected: (marker) => {
			dispatch(ReduxActions.setMarkerSelected(marker))
		},
		setMarkerViewVisible: () => {
			dispatch(ReduxActions.setMarkerViewVisible())
		},
		disableGestures: (value) => {
			dispatch(ReduxActions.disableGestures(value))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MapContainer)