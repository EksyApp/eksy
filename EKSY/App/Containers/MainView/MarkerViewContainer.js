import React, {Component} from "react";
import * as ReduxActions from "../../Actions/index";
import {connect} from "react-redux";
import MarkerViewComponent from "../../Components/MainView/MarkerViewComponent";



export class MarkerViewContainer extends Component {
	
	render() {
		return (
				<MarkerViewComponent
				marker = {this.props.marker}
				markerViewVisible = {this.props.markerViewVisible}
				setMarkerViewHidden = {this.props.setMarkerViewHidden}
				/>
		)
	}
}

const mapStateToProps = (state) => {
	return {
		marker: state.markers.markerSelected,
		markerViewVisible: state.ui.markerView.markerViewVisible,
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		setMarkerViewHidden: () => {
			dispatch(ReduxActions.setMarkerViewHidden())
			dispatch(ReduxActions.disableGestures(false))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MarkerViewContainer)
