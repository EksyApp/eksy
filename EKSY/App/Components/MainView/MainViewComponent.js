import React, {Component} from 'react'
import MenuButton from '../../Components/Common/MenuButton'
import {View} from 'react-native'
import {connect} from 'react-redux'
import MarkerViewContainer from "../../Containers/MainView/MarkerViewContainer";
import BottomSheetContainer from "../../Containers/MainView/BottomSheetContainer";
import MapContainer from "../../Containers/MainView/MapContainer";


class MainViewComponent extends Component {
	
	render() {
		return (
				<View>
					<MapContainer />
					<View style={styles.panelContainer}>
						<BottomSheetContainer />
						<MarkerViewContainer/>
					</View>
					<MenuButton onPress={() => {this.props.menuButtonPress()}}/>
				</View>
		)
	}
}


const styles = StyleSheet.create({
	panelContainer: {
		position: 'absolute',
		top: 0,
		bottom: 0,
		left: 0,
		right: 0,
		flex: 1,
	}
})

export default MainViewComponent