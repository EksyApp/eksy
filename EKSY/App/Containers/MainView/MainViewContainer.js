import React, {Component} from 'react'
import * as ReduxActions from '../../Actions/index'
import {connect} from 'react-redux'
import MainViewComponent from "../../Components/MainView/MainViewComponent";
import PropTypes from 'prop-types'

//Renders the main view and handles it's logic
export class MainViewContainer extends Component {
	
	render() {
		return (
				<MainViewComponent
						menuButtonPress = {this.props.menuButtonPress}
				/>
		)
	}
}

MainViewContainer.propTypes = {
	menuButtonPress: PropTypes.func
}

const mapStateToProps = (state) => {
	return {
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		menuButtonPress: () => {
			dispatch(ReduxActions.drawerOpen())
		},
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MainViewContainer)
