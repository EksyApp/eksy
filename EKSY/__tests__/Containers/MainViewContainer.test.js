import React from 'react'
import { MainViewContainer } from '../../App/Containers/MainView/MainViewContainer'
import renderer from 'react-test-renderer'

jest.mock('../../App/Containers/MainView/Map/MapContainer', () => {
	return require('react-native').View
})

jest.mock('../../App/Containers/MainView/BottomSheet/BottomSheetContainer', () => {
	return require('react-native').View
})

jest.mock('../../App/Containers/MainView/MarkerModal/MarkerModalContainer', () => {
	return require('react-native').View
})

test('MainViewContainer renders correctly', () => {
	const tree = renderer.create(
			<MainViewContainer
					menuButtonPress={() => {}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});