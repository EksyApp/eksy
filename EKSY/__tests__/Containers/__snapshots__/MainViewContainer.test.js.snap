// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainViewContainer renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View />
    <View />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          Object {
            "height": 50,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 50,
            "zIndex": 1,
          },
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 35,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "color": "rgb(101,114,122)",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î—’
    </Text>
  </View>
</View>
`;
